pkg/sysfs/syspath.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sysfs
+
+import (
+	"fmt"
+	"github.com/openebs/node-disk-manager/blockdevice"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	// BlockSubSystem is the key used to represent block subsystem in sysfs
+	BlockSubSystem = "block"
+	// NVMeSubSystem is the key used to represent nvme subsystem in sysfs
+	NVMeSubSystem = "nvme"
+	// sectorSize is the sector size as understood by the unix systems.
+	// It is kept as 512 bytes. all entries in /sys/class/block/sda/size
+	// are in 512 byte blocks
+	sectorSize int64 = 512
+)
+
+var sysFSDirectoryPath = "/sys/"
+
+// getDeviceSysPath gets the syspath struct for the given blockdevice.
+// It is generated by evaluating the symlink in /sys/class/block.
+func getDeviceSysPath(devicePath string) (string, error) {
```

Comment:
@akhilerm how will NVMe devices be handled by this function?

Reply:
Yes @harshthakur9030 . All block devices will have an entry in `/sys/class/block/`. Only the symlink it points to will be different.